### PR TITLE
Open redirect in opencart

### PR DESCRIPTION
**Pull request by kajalNair**

### CODE REVIEW

File: bounties/packagist/opencart/opencart/2/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "I": "H",
+    "A": "N",
+    "E": "",
+    "RL": "",
+    "RC": "",
+    "Score": "6.4"
+  },
+  "CVEs": [""],
+  "Repository": {
+    "URL": "",
```
Comment (Mik317) [inline bounties/packagist/opencart/opencart/2/vulnerability.json:38]: Add the repo please :smile:

File: bounties/packagist/opencart/opencart/2/vulnerability.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "PackageVulnerabilityID": "2",
+  "DisclosureDate": "9-12-2020",
+  "AffectedVersionRange": "",
```
Comment (Mik317) [inline bounties/packagist/opencart/opencart/2/vulnerability.json:4]: Please define the `versions` affected field :smile:

**Review by Mik317 — COMMENTED** [review]
Hi @kajalNair :smile:,
I'm not able to get the `open redirect` work :cry:

Could you please share the `POST` request you make to get the redirection?
I used this one:

```http
POST /admin/index.php?route=common/login HTTP/1.1
Host: 0.0.0.0:8001
Content-Length: 370
Cache-Control: max-age=0
Origin: http://localtest.me:8001
Upgrade-Insecure-Requests: 1
Content-Type: multipart/form-data; boundary=----WebKitFormBoundaryIQtJVAfL42IxBJOf
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.162 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9,it;q=0.8,ru;q=0.7
Connection: close

------WebKitFormBoundaryIQtJVAfL42IxBJOf
Content-Disposition: form-data; name="username"

admin
------WebKitFormBoundaryIQtJVAfL42IxBJOf
Content-Disposition: form-data; name="password"

testfortest123A!X
------WebKitFormBoundaryIQtJVAfL42IxBJOf
Content-Disposition: form-data; name="redirect"

http://evil.com
------WebKitFormBoundaryIQtJVAfL42IxBJOf--

```
but it didn't work since some `validations` are made.

Cheers,
Mik

File: bounties/packagist/opencart/opencart/2/vulnerability.json (hunk 1 above)
Comment (kajalNair) [reply bounties/packagist/opencart/opencart/2/vulnerability.json]: done.

File: bounties/packagist/opencart/opencart/2/vulnerability.json (hunk 2 above)
Comment (kajalNair) [reply bounties/packagist/opencart/opencart/2/vulnerability.json]: done

**Comment by kajalNair** [comment]
Hi Michele,

The application uses strpos to compare the host address. The strpos() function could be bypassed using www.opencart.com.evil.com domain. Can you check with this domain.

**Review by Mik317 — CHANGES_REQUESTED** [review]
Hi @kajalNair :smile:,
I'm sorry but I'm still unable to validate the issue :cry:

I tried as you explained but it didn't work. I tried also putting the `hostname` of my `local installation` but it didn't work.
Then, I checked the `protection` you mentioned and I'm not sure it's bypassable: `HTTP_SERVER` indexes to `hostname/admin` which can't be bypassed as far as I know, since the `path` segment is already initialized in the `redirection` process.

Screenshot from 2020-09-14 15-28-37

Could you send me the exact request you made to achieve `open redirect` ?

Cheers,
Mik

**Comment by kajalNair** [comment]
Hi Michelle,

I will recheck and revert back.
What about the other instance?

Kajal

**Comment by 418raj** [comment]
Hi,

Unfortunately I'm unable to reproduce this too so will close this PR for now - feel free to make a new one though for it if you have more instructions for the PoC.

Thanks

**Pull request closed without merging**